db/columns/base.py

```diff
@@ -138,12 +138,29 @@ def column_index(self):
                 self.engine
             )
 
+    @property
+    def column_attnum(self):
+        """
+        Get the ordinal index of this column in its table, if it is
+        attached to a table that is associated with the column's engine.
+        """
+        if (
+                self.engine is not None
+                and self.table_ is not None
+                and inspect(self.engine).has_table(self.table_.name, schema=self.table_.schema)
+        ):
```

Comment:
I think this would be a bit easier to read if you put `inspect(self.engine).has_table(self.table_.name, schema=self.table_.schema)` in a variable and maybe the other two conditionals as well. Then you could put the if statement in a single line.
```suggestion
        engine_exists = self.engine is not None
        table_exists = self.table_ is not None
        engine_has_table = inspect(self.engine).has_table(self.table_.name, schema=self.table_.schema)
        if engine_exists and table_exists and engine_has_table:
```